artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerTarget.java

```diff
@@ -476,4 +487,48 @@ public void postAcknowledge(MessageReference ref, AckReason reason) {
    public void sendMessage(Message message, RoutingContext context, List<MessageReference> refs) {
    }
 
+   // I need a supress warning annotation here
+   // because errorProne is issuing an error her, however I really intend to compare PageACK against PagedReference
+   @SuppressWarnings("ComparableType")
```

Comment:
I thought it was nice to use Comparable.. which was what I wanted.. I didn't anticipate the warnings...

I will move to IntFunction.

Reply:
I need the opposite of IntFunction though... I need a Function<PageReference,Integer>

I will create my own to avoid the boxing. at least until I could find an equivalent on JDK.

Reply:
Oops. I was thinking of what ToIntFunction does.

Reply:
ahh.. yes.. ToIntFunction.. that's what I was looking for.. :)